server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -713,6 +715,37 @@ def __init__(
             sliding_window=sliding_window,
         )
 
+        if kv_cache_dtype == "fp8":
+            self.kv_cache_dtype = torch.uint8
+        else:
+            self.kv_cache_dtype = self.dtype
+
+        if kv_cache_dtype == "fp8" and SYSTEM == "rocm":
+            logger.info(f"Using KV cache data type: {kv_cache_dtype}")
+            # Currently scaled KV cache is only enabled on ROCm
+            if quantization_param_path is not None:
+                if callable(getattr(self.model, "load_kv_cache_scales", None)):
+                    self.model.load_kv_cache_scales(quantization_param_path)
+                else:
+                    raise RuntimeError(
+                        "Using FP8 KV cache and scaling "
+                        "factors provided but model "
+                        f"{self.model.__class__} does not "
+                        "support loading scaling factors."
+                    )
+            else:
+                logger.info(
+                    "Using FP8 KV cache but no scaling factors "
+                    "provided. Defaulting to scaling factors of 1.0. "
+                    "This may lead to less accurate results!"
+                )
+        elif quantization_param_path is not None:
```

Comment:
This should be handled all the way back in the launcher (again ideally directly by clap, if not possible manually in the rust part). 

And it should be hard error, not a soft one (paramters sent by the user don't make sense, we never silently ignore.)